projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistry.scala

```diff
@@ -15,52 +16,65 @@ import com.lightbend.lagom.internal.projection.ProjectionRegistry._
 import com.lightbend.lagom.internal.cluster.ClusterDistribution
 import com.lightbend.lagom.internal.cluster.ClusterDistributionSettings
 import com.lightbend.lagom.internal.projection.ProjectionRegistryActor.DesiredState
-import com.lightbend.lagom.internal.projection.ProjectionRegistryActor.GetDesiredState
+import com.lightbend.lagom.internal.projection.ProjectionRegistryActor.GetState
+import com.lightbend.lagom.internal.projection.ProjectionRegistryActor.State
 
 import scala.concurrent.ExecutionContext
 import scala.concurrent.Future
 import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
 
 @ApiMayChange
 object ProjectionRegistry {
 
   sealed trait WorkerStatus
-  case object Stopped extends WorkerStatus
-  case object Started extends WorkerStatus
+  sealed trait Stopped extends WorkerStatus
+  case object Stopped  extends Stopped
+  sealed trait Started extends WorkerStatus
+  case object Started  extends Started
 
-  sealed trait StateRequest
-  case object Stop  extends StateRequest
-  case object Start extends StateRequest
+  sealed trait StateRequest {
+    def workerName: String
+  }
+  case class Stop(workerName: String)  extends StateRequest
+  case class Start(workerName: String) extends StateRequest
 
   @ApiMayChange
-  case class ProjectionWorker(name: String, status: WorkerStatus)
+  case class ProjectionWorker(name: String, status: StateRequest)
```

Comment:
Hmm, I'm confused. 

Is `StateRequest` a kind of message/command? If so, we should avoid using it in the model. 

If it's a command, I find the current name correct. If it's part of the model, then `RequestedState` is better.

If I understand it correctly, this is a real request type. In that case, I suggest the following for the model.

```scala
case class ProjectionWorker(name: String, requestedState: WorkerStatus, observedState: WorkerStatus)
```

Reply:
Yeah, I'm reusing the command (`StateRequest`) in the model. Will review.